shell/common/platform_util_linux.cc

```diff
@@ -408,15 +410,7 @@ bool PlatformTrashItem(const base::FilePath& full_path, std::string* error) {
 }  // namespace internal
 
 void Beep() {
-  // echo '\a' > /dev/console
-  FILE* fp = fopen("/dev/console", "a");
-  if (fp == nullptr) {
-    fp = fopen("/dev/tty", "a");
-  }
-  if (fp != nullptr) {
-    fprintf(fp, "\a");
-    fclose(fp);
-  }
+  gdk_display_beep(gdk_display_get_default());
```

Comment:
`gdk_display_beep` is missing stub declaration. Can you create `electron_gdk.sigs` in https://github.com/electron/electron/tree/main/shell/browser/ui and add it to https://github.com/electron/electron/blob/43ed763f181ff648030fb498e7c538decd51c352/BUILD.gn#L87. You can then use `#include electron/electron_gtk_stubs.h` which will also bring in https://github.com/electron/electron/blob/main/shell/browser/ui/electron_gtk.fragment

`gdk_display_get_default` is declared in upstream https://source.chromium.org/chromium/chromium/src/+/main:ui/gtk/gdk.sigs;l=1.

Reply:
Also realized we don't use https://github.com/electron/electron/blob/main/shell/browser/ui/electron_gtk.sigs since the file dialog refactor, feel free to remove it when you are making the above changes.

Reply:
@deepak1556 SGTM, thanks for the links and the walkthrough!

We went through this a month or so back with a GdkPixbuf change, but I (incorrectly?) thought this only applied to the gdk-pixbuf library, not larger gdk overall .

Is this the process for any new GLIB / GDK / GTK calls?

Reply:
Yup the process is same for any new gdk / gtk / gio calls which are not covered by the [stubs](https://source.chromium.org/chromium/chromium/src/+/main:ui/gtk) (*.sig) files in upstream. All of these are runtime loaded libraries, you can see their initialization here https://source.chromium.org/chromium/chromium/src/+/main:ui/gtk/gtk_compat.cc;l=95-102.

Reply:
> You can then use #include electron/electron_gtk_stubs.h which will also bring in https://github.com/electron/electron/blob/main/shell/browser/ui/electron_gtk.fragment, avoids having to repeat #include <gtk/gtk.h> separately.

This fragment file gets fed into GN as an [`extra_header`](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/tools/generate_stubs/rules.gni#11), so it's inserted into `gen/electron/electron_gtk_stubs.cc` rather than `electron_gtk_stubs.h`. :man_shrugging:  I think we still need to include GTK?